x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/storage/ReactiveStorageDeciderService.java

```diff
@@ -722,6 +798,8 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             builder.field("reason", reason);
             builder.field("unassigned", unassigned);
             builder.field("assigned", assigned);
+            builder.field("unassignedShardIds", unassignedShardIds);
```

Comment:
We use underscores in our json representation: `s/unassignedShardIds/unassigned_shard_ids/`

Reply:
I'm not sure if `unassigned` is a right term for this API as it is possible that a shard that should move from hot to warm remains assigned to a hot node, but for the purposes of this decider we consider it as `unassigned` to trigger the tier creation.  But we use these terms already, so we can leave as it is.

Reply:
I was thinking that it might make sense to include some information around the allocation decision for these shards so it is easier to debug issues around that. The downside is that responses can get too large.

Reply:
Done!

Reply:
I also added a test for the JSON representation of `ReactiveReason`

Reply:
> I was thinking that it might make sense to include some information around the allocation decision 

I agree to finding at least one shard to get more info for. But I'd like to keep that separate to this PR, this has value on it's own.